app/models/weam.rb

```diff
@@ -40,15 +46,15 @@ class Weam < ActiveRecord::Base
     'ope' => { column: :ope, converter: OpeConverter }
   }.freeze
 
-  validates :facility_code, :institution, :institution_type, presence: true
+  validates :facility_code, :institution, :institution_type_name, presence: true
   validates :bah, numericality: true, allow_blank: true
 
   before_validation :derive_dependent_columns
 
   # Computes all fields that are dependent on other fields. Called in validation because
   # activerecord-import does not engage callbacks when saving
   def derive_dependent_columns
-    self.institution_type = derive_type
+    self.institution_type_name = derive_type
```

Comment:
I think it would be better to do this in an initializer or a class method on the object that initializes a new version of the model or at the very least this should be a private method invoked by an initializer.